src/main/java/com/google/ehub/utility/BlobstoreURLUtility.java

```diff
@@ -0,0 +1,60 @@
+package com.google.ehub.utility;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Utility class used to find Blobstore upload URL from form element.
+ */
+public final class BlobstoreURLUtility {
+  /**
+   * Finds the URL that points to the uploaded file in Blobstore.
+   *
+   * @param request the request sent by the upload form submission
+   * @param formUploadElem the ID of the element in the form that selects the file to upload
+   * @return the upload URL of the submitted file wrapped in an {@link Optional}
```

Comment:
explain here why it might be empty

Reply:
I added documentation for cases when Optional would be empty in last commit.